### PR TITLE
Fix replacing markers

### PR DESCRIPTION
**Pull request by TFP-dev**
Based on #2794. Instead of adding ```engineReinitMarkers```, it executes 3D markers reinit function when replacing or restoring models with id 1316 (ring), 1317 (cylinder and checkpoint) and 1559 (arrow).

markers

### Test resource
[replace_markers_test_resource.zip](https://github.com/multitheftauto/mtasa-blue/files/11686873/replace_markers_test_resource.zip)

### CODE REVIEW

File: Client/mods/deathmatch/logic/CClientDFF.cpp [hunk 1]
```diff
@@ -280,6 +284,10 @@ void CClientDFF::InternalRestoreModel(unsigned short usModel)
             g_pGame->GetRenderWare()->DestroyDFF(pInfo->pClump);
         MapRemove(m_LoadedClumpInfoMap, usModel);
     }
+
+    // 'Restream' 3D markers
+    if (usModel == 1316 || usModel == 1317 || usModel == 1559)
+        g_pClientGame->ReinitMarkers();
```
Comment (TheNormalnij) [inline Client/mods/deathmatch/logic/CClientDFF.cpp]: Can be better DI here:
```c++
else if (CClientMarkerManager::IsMarkerModel(usModel)) {
    m_pManager->GetMarkerManager()->ReinitMarkers();
}
```

File: Client/mods/deathmatch/logic/CClientDFF.cpp [hunk 2]
```diff
@@ -109,6 +109,10 @@ bool CClientDFF::ReplaceModel(unsigned short usModel, bool bAlphaTransparency)
 
     bool bResult = DoReplaceModel(usModel, bAlphaTransparency);
 
+    // 'Restream' 3D markers
+    if (usModel == 1316 || usModel == 1317 || usModel == 1559)
+        g_pClientGame->ReinitMarkers();
+
```
Comment (TheNormalnij) [inline Client/mods/deathmatch/logic/CClientDFF.cpp]: Should be in `DoReplaceModel` method with better DI.

Comment (TFP-dev) [reply Client/mods/deathmatch/logic/CClientDFF.cpp]: Done

File: Client/mods/deathmatch/logic/CClientDFF.cpp (hunk 1 above)
Comment (TFP-dev) [reply Client/mods/deathmatch/logic/CClientDFF.cpp]: Done

File: Client/mods/deathmatch/logic/CClientDFF.cpp [hunk 3]
```diff
@@ -164,6 +164,15 @@ bool CClientDFF::DoReplaceModel(unsigned short usModel, bool bAlphaTransparency)
         {
             return ReplacePedModel(pClump, usModel, bAlphaTransparency);
         }
+        else if (CClientMarkerManager::IsMarkerModel(usModel))
+        {
+            bool bResult = ReplaceObjectModel(pClump, usModel, bAlphaTransparency);
+
+            // 'Restream' 3D markers
+            g_pClientGame->ReinitMarkers();
```
Comment (botder) [inline Client/mods/deathmatch/logic/CClientDFF.cpp]: Only do this when `bResult` is true.

File: Client/mods/deathmatch/logic/CClientDFF.cpp [hunk 4]
```diff
@@ -164,6 +164,15 @@ bool CClientDFF::DoReplaceModel(unsigned short usModel, bool bAlphaTransparency)
         {
             return ReplacePedModel(pClump, usModel, bAlphaTransparency);
         }
+        else if (CClientMarkerManager::IsMarkerModel(usModel))
+        {
+            bool bResult = ReplaceObjectModel(pClump, usModel, bAlphaTransparency);
```
Comment (botder) [inline Client/mods/deathmatch/logic/CClientDFF.cpp]: Rename `bResult` to `wasReplaced`.

**Review by botder — DISMISSED** [review]

**Review by botder — APPROVED** [review]

**Pull request merged**